src/metabase/sync.clj

```diff
@@ -76,3 +82,32 @@
                                              (sync-util/name-for-logging field))
         (fingerprint/refingerprint-field field))
       :sync/no-connection)))
+
+(s/defn sync-new-table!
+  "Given the name and schema for a new table (one that exists in the user's db but not as a Metabase table), add the
+  table to Metabase and sync the new table's metadata."
+  [db :- i/DatabaseInstance
+   {:keys [schema-name table-name]} :- {:schema-name s/Str
+                                        :table-name  s/Str}]
+  (let [normalize-table-desc (fn [{:keys [schema name]}]
+                               (cond-> {:name (u/lower-case-en name)}
+                                       schema
+                                       (assoc :schema (u/lower-case-en schema))))
+        target-table #{(normalize-table-desc {:schema schema-name :name table-name})}]
+    (if (some->> db
```

Comment:
Couple thoughts:
1. I generally like to `let` bind results that would come back from `(driver/describe-database ...)`. This isn't any kind of rule just something i tend to do. Often helps me log or tap> it when developing. Hard to get a picture of what's going on otherwise.
2. I think we might have a logic bug due to `normalize-table-desc`. We probably want to identify the user input with existing table information with `normalize-table-desc`. But we really want to save the exact value that came from `describe-database` into our app db, and not whatever random casing the user typed that happened to be equal to what the driver says under `u/lower-case-en`. So I think the `(map normalize-table-desc)` is a bad call and we want to capture the map which is equal. So it would be more like 

```clojure
(s/defn sync-new-table!
  "Given the name and schema for a new table (one that exists in the user's db but not as a Metabase table), add the
  table to Metabase and sync the new table's metadata."
  [db :- i/DatabaseInstance
   {:keys [schema-name table-name]} :- {:schema-name s/Str
                                        :table-name  s/Str}]
  ;; shorter name for local `normalize`
  (let [normalize (fn [{:keys [schema name]}]
                    (cond-> {:name (u/lower-case-en name)}
                      schema
                      (assoc :schema (u/lower-case-en schema))))
        target-table #{(normalize {:schema schema-name :name table-name})}
        ;; let-bind the db results
        {db-tables :tables} (driver/describe-database (driver.u/database->driver db) db)]
    ;; make sure to bind to the info that comes back from describe-database
    (if-let [new-table (some (fn [db-table]
                               ;; call `normalize` inside of the `some` and not `map` so we have the original
                               (when (target-table (normalize db-table))
                                 db-table))
                             db-tables)]
      (if (->> (sync-tables/sync-tables-and-database! db {:tables #{new-table}})
               :updated-tables
               pos?)
        (when-some [table (db/select-one 'Table
                                         :db_id (u/the-id db)
                                         :schema (:schema new-table)
                                         :name (:name new-table))]
          (sync-table! table))
        (log/info (trs "Table ''{0}'' already exists." table-name)))
      (log/info (trs "Table ''{0}'' does not exist or you do not have permission to view it." table-name)))))
```

We also need to not use `sync-tables/sync-tables-and-database!`. We are abusing it and not thinking about what it is doing. I'll post this repl output and see if anything stands out:

```clojure
sync=> (let [db (db/select-one 'Database :id 1)]
         (sync-tables/sync-tables-and-database! db {:tables #{{:name "ORDERS" :schema "PUBLIC"}}}))
{:updated-tables 7, :total-tables 8}
```

I told it to sync the `"ORDERS"` table. But note that it returns that it has updated 7 tables. It has marked all other tables inactive because it assumes that the metadata passed in is the state of the world, meaning the only table that exists in that database is the ORDERS table and it deactivates all of the other tables.

```clojure
sync=> (map #(select-keys % [:id :name :schema :active]) (db/select 'Table :db_id 1))
({:id 4, :name "FEEDBACK", :schema "PUBLIC", :active false}
 {:id 2, :name "ORDERS", :schema "PUBLIC", :active true}
 {:id 5, :name "PEOPLE", :schema "PUBLIC", :active false}
 {:id 6, :name "INVOICES", :schema "PUBLIC", :active false}
 {:id 7, :name "ACCOUNTS", :schema "PUBLIC", :active false}
 {:id 8, :name "REVIEWS", :schema "PUBLIC", :active false}
 {:id 1, :name "PRODUCTS", :schema "PUBLIC", :active false}
 {:id 3, :name "ANALYTIC_EVENTS", :schema "PUBLIC", :active false})
```

Reply:
Updated as described and refactored to just call `create-or-reactivate-table!`.